components/Swap/Withdraw/Wallet/WalletTransfer/TransferErc20.tsx

```diff
@@ -102,8 +103,12 @@ const TransferErc20Button: FC<TransferERC20ButtonProps> = ({
         hash: contractWrite?.data?.hash || savedTransactionHash,
         onSuccess: async (trxRcpt) => {
             setApplyingTransaction(true)
-            setSwapPublishedTx(swapId, PublishedSwapTransactionStatus.Completed, trxRcpt.transactionHash);
+            setSwapTransaction(swapId, PublishedSwapTransactionStatus.Completed, trxRcpt.transactionHash);
             setApplyingTransaction(false)
+        },
+        onError: async (err) => {
+            console.log(err.message, "err222")
+            setSwapTransaction(swapId, PublishedSwapTransactionStatus.Error, "", err.message);
```

Comment:
Transaction ID is lost here. 
So later when you try to get the transaction for this swap ID you'll get the one that's been set when the transaction was pending on line 88 here.